app/services/open_library_service.rb

```diff
@@ -0,0 +1,17 @@
+class OpenLibraryService
+  def book_info(isbn)
+    url = 'https://openlibrary.org/api/books?bibkeys=ISBN:' + isbn + '&format=json&jscmd=data'
+    @data = (JSON.parse HTTParty.get(url).to_s).values[0]
+    @response = {}
+    generate_response unless @data.nil?
```

Comment:
and here just do
`generate_response(data) unless data.nil?`

Reply:
Or maybe even just `generate_response(data)`
and have a `return {} unless data.present?` as the first line of generate_response